### PR TITLE
107 configure ci infrastructure

### LINKED ISSUE #107: Configure CI Infrastructure
- [ ]  add commit linter workflow
- [ ]  add linter workflow

### PR DESCRIPTION
**Pull request by dstallenberg**
Closes #107

### CODE REVIEW

File: src/analysis/static/graph/drawGraph.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (mitchellolsthoorn) [inline src/analysis/static/graph/drawGraph.ts:1]: Do these need to be above the copyright?

File: src/search/operators/ranking/FastNonDomSorting.ts [hunk 2]
```diff
@@ -48,14 +48,14 @@ export function fastNonDomSorting<T extends Encoding>(
       let pDominatesQ = true;
       let qDominatesP = true;
       for (const key of objectiveFunctions) {
-        if (p.getDistance(key)! === 0 && q.getDistance(key)! === 0) {
+        if (p.getDistance(key) === 0 && q.getDistance(key) === 0) {
```
Comment (mitchellolsthoorn) [inline src/search/operators/ranking/FastNonDomSorting.ts:51]: What did the exclamation mark do?

Comment (dstallenberg) [reply src/search/operators/ranking/FastNonDomSorting.ts:51]: ensure "not null" but other changes I made, already ensure "not null"

File: .github/workflows/verify.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    strategy:
+      # Keep running other versions when a job fails
+      fail-fast: false
+
+      matrix:
+        # Run the pipeline on all the currently supported OS versions
+        os: [ubuntu-latest]
+
+        # Run the pipeline on all the currently supported LTS versions and the upcoming version
+        node-version: [18, 19]
```
[the remark this answers is not in the document]
Comment (dstallenberg) [reply .github/workflows/verify.yml]: Nope, fixed

File: src/analysis/static/graph/drawGraph.ts (hunk 1 above)
Comment (dstallenberg) [reply src/analysis/static/graph/drawGraph.ts]: Nope, fixed

File: src/config.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
[the remark this answers is not in the document]
Comment (dstallenberg) [reply src/config.ts]: Nope, fixed

**Review by mitchellolsthoorn — APPROVED** [review]

**Pull request merged**